src/tools/rbd_mirror/image_map/Policy.cc

```diff
@@ -130,12 +131,52 @@ void Policy::remove_instances(const std::vector<std::string> &instance_ids,
   m_dead_instances.insert(instance_ids.begin(), instance_ids.end());
 }
 
+// return the next transitions state depending on the current
+// operation mode
+StateTransition::State Policy::get_transition_state(const ActionState &action_state) {
+  dout(20) << dendl;
+
+  StateTransition::State state;
+  if (!action_state.in_rollback || !action_state.transition.error_state) {
+    state = action_state.transition.next_state;
+  } else {
+    state = action_state.transition.error_state.get();
+  }
+
+  return state;
+}
+
+// return a state to be used as the final state when finishing
+// a transition.
+StateTransition::State Policy::get_final_state(StateTransition::State state) {
+  if (state == StateTransition::STATE_REMOVE_MAPPING) {
+    return StateTransition::STATE_UNASSIGNED;
+  }
+
+  dout(20) << ": final_state=" << state << dendl;
+  return state;
+}
+
 // new actions are always started from a stable (idle) state since
 // actions either complete successfully ending up in an idle state
 // or get aborted due to peer being blacklisted.
 void Policy::start_next_action(const std::string &global_image_id) {
```

Comment:
Nit: seems odd that you have two functions w/ the same name that do two different things (i.e. this one loops through all available actions). Perhaps the next method should be named something like ```process_next_state```?